WalletWasabi.Tests/UnitTests/BranchAndBoundSelectionTests.cs

```diff
@@ -0,0 +1,51 @@
+using NBitcoin;
+using System.Collections.Generic;
+using WalletWasabi.BranchNBound;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests
+{
+	public class BranchAndBoundSelectionTests
+	{
+		private Random Random { get; } = new();
+
+		[Fact]
+		public void RandomizedTest()
+		{
+			var utxos = GenList();
+			var selector = new BranchAndBound(utxos);
+			ulong target = Money.Satoshis(100000000);
+
+			var successful = selector.TryGetExactMatch(target, out List<Money> selectedCoins);
+
+			Assert.True(successful);
+			Assert.NotNull(selectedCoins);
+			Assert.Equal(target, (ulong)selector.CalcEffectiveValue(selectedCoins));
+		}
+
```

Comment:
Test ideas:

* One test might be "Have 5 coins and all of them are needed and sum of them is the target".
* Another test should be "We have 1000 of coins and their sum is lower than target" to verify that we finish immediately.